tenable/io/v3/vm/credentials.py

```diff
@@ -0,0 +1,260 @@
+    def _permissions_constructor(
+            self, permissions: List[Union[Tuple, Dict]]) -> List[Dict]:
+        '''
+        Validates and/or transforms thew permissions items into the desired
+        format.  If a dict it will validate.  If a tuple, will convert.
+        '''
+        permissions_data = list()
+        permission_schema = CredentialsPermissionsSchema()
+        for permission in permissions:
+            perm = permission_schema.dump(
+                permission_schema.load(permission))
+            permissions_data.append(perm)
+        return permissions_data
+
+    def create(self, cred_name: str, cred_type: str,
+               description: Optional[str] = None,
+               permissions: Optional[List] = None,
+               **settings: Optional[Dict]) -> str:
+        '''
+        Creates a new managed credential.
+
+        :devportal:`credentials: create <credentials-create>`
+
+        Args:
+            cred_name (str):
+                The name of the credential.
+            cred_type (str):
+                The type of credential to create.  For a list of values refer
+                 to
+                the output of the :py:meth:`types() <CredentialsAPI.types>`
+                method.
+            description (str, optional):
+                A description for the credential.
+            permissions (list, optional):
+                A list of permissions (in either tuple or native dict format)
+                detailing whom is allowed to use or edit this credential set.
+                For the dictionary format, refer to the API docs.  The tuple
+                format uses the customary ``(type, perm, uuid)`` format.
+
+                Examples:
+                    - ``('user', 32, user_uuid)``
+                    - ``('group', 32, group_uuid)``
+                    - ``('user', 'use', user_uuid)``
+
+            **settings (dict, optional):
+                Additional keywords passed will be added to the settings dict
+                within the API call.  As this dataset can be highly variable,
+                it will not be validated and simply passed as-is.
+
+        Returns:
+            :obj:`str`:
+                The UUID of the newly created credential.
+
+        Examples:
+            >>> group_id = '00000000-0000-0000-0000-000000000000'
+            >>> tio.v3.vm.credentials.create('SSH Account', 'SSH',
+            ...     permissions=[('group', 'use', group_id)],
+            ...     username='user1',
+            ...     password='sekretsquirrel',
+            ...     escalation_account='root',
+            ...     escalation_password='sudopassword',
+            ...     elevate_privileges_with='sudo',
+            ...     bin_directory='/usr/bin',
+            ...     custom_password_prompt='')
+        '''
+        permissions_data = permissions
+        if permissions and isinstance(permissions, list):
+            permissions_data = self._permissions_constructor(permissions)
```

Comment:
If permissions is not a list, then the _permissions_constructor will not be called? What will be passed in permissions in that case?

Reply:
default value is None. so if permissions is not a list and also not None, then it will be validated in create_schema.